feat(Liquid.Cache.Memory): Refactoring of Liquid.Cache.Memory

lucianareginalino opened the pull request:
This implementation just extends  Microsoft.Extensions.Caching.Memory DI by including Liquid.Cache decorator registry.
Resolves #140
Signed-off-by: Luciana Regina Lino <luciana.regina.lino@avanade.com>

guilhermeluizsp commented on test/Liquid.Base.Tests/WorkbenchTests.cs:
```diff
@@ -0,0 +1,425 @@
+            service.Received(1).Initialize();
+        }
+
+        private class MockMediaStorage : ILightMediaStorage
```
Since you chose to use NSubstitute, you could leverage the [Substitute.For](https://nsubstitute.github.io/help/creating-a-substitute/) method instead of manually implementing a class for every dependency.

bruno-brant replied:
Actually, I can't. Unless I use reflection to create those stubs, I can't use reflection because there's no overload for UseService that accepts a instance of Type. All I'm left with is a generic method that requires me to pass a class token.

Again, I could use reflection to call the method but I think that would make the test more obscure.

guilhermeluizsp replied:
Of course, sorry for the overlook.
Thinking about it, we could turn that into a feature, as those methods should be able to accept instances as well. But that's for another PR. Thanks for pointing that out anyways.

bruno-brant commented:
Yeah, I was also considering whether they could be extension methods. In any case, I think we should consider just removing workbench.

lucianareginalino committed:
feat(Liquid.Cache.Memory): Refactoring of Liquid.Cache.Memory to make it more compatible with dotnet framework.
This implementation just extends  Microsoft.Extensions.Caching.Memory DI by including Liquid.Cache decorator registry.
Resolves #140
Signed-off-by: Luciana Regina Lino <luciana.regina.lino@avanade.com>

lucianareginalino committed:
feat(liquid-ci-cd-cache-memory.yml): Creates Liquid.Cache.Memory CI/CD Action.
Signed-off-by: Luciana Regina Lino <luciana.regina.lino@avanade.com>
